x-pack/plugins/siem/server/endpoint/routes/resolver/queries/stats.ts

```diff
@@ -5,13 +5,19 @@
  */
 import { SearchResponse } from 'elasticsearch';
 import { ResolverQuery } from './base';
-import { ResolverEvent } from '../../../../../common/endpoint/types';
+import { ResolverEvent, EventStats } from '../../../../../common/endpoint/types';
 import { JsonObject } from '../../../../../../../../src/plugins/kibana_utils/public';
 import { AggBucket } from '../utils/pagination';
 
 export interface StatsResult {
   alerts: Record<string, number>;
-  events: Record<string, number>;
+  events: Record<string, EventStats>;
+}
+
+interface CategoriesAgg extends AggBucket {
+  categories?: {
```

Comment:
:grey_question: It seems counterintuitive to have an extending type with one property that is optional. Are you sure that should be optional?

Reply:
Yeah it's optional because in the case where no results are found in the stats query the aggregations won't exist. Here's an example:

<details> <summary> With results </summary>

`title` aggregation exists

```
{
  "took" : 10,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 5,
      "relation" : "eq"
    },
    "max_score" : 0.0,
    "hits" : [
      {
        "_index" : "test-agg",
        "_type" : "_doc",
        "_id" : "1",
        "_score" : 0.0,
        "_source" : {
          "entity_id" : "1",
          "category" : [
            "authentication",
            "session"
          ]
        }
      },
      {
        "_index" : "test-agg",
        "_type" : "_doc",
        "_id" : "2",
        "_score" : 0.0,
        "_source" : {
          "entity_id" : "1",
          "category" : [
            "authentication",
            "session"
          ]
        }
      },
      {
        "_index" : "test-agg",
        "_type" : "_doc",
        "_id" : "3",
        "_score" : 0.0,
        "_source" : {
          "entity_id" : "1",
          "category" : "session"
        }
      },
      {
        "_index" : "test-agg",
        "_type" : "_doc",
        "_id" : "4",
        "_score" : 0.0,
        "_source" : {
          "entity_id" : "2",
          "category" : "super_title"
        }
      },
      {
        "_index" : "test-agg",
        "_type" : "_doc",
        "_id" : "5",
        "_score" : 0.0,
        "_source" : {
          "entity_id" : "2",
          "category" : "super_title"
        }
      }
    ]
  },
  "aggregations" : {
    "genres" : {
      "doc_count_error_upper_bound" : 0,
      "sum_other_doc_count" : 0,
      "buckets" : [
        {
          "key" : "1",
          "doc_count" : 3,
          "title" : {
            "doc_count_error_upper_bound" : 0,
            "sum_other_doc_count" : 0,
            "buckets" : [
              {
                "key" : "session",
                "doc_count" : 3
              },
              {
                "key" : "authentication",
                "doc_count" : 2
              }
            ]
          }
        },
        {
          "key" : "2",
          "doc_count" : 2,
          "title" : {
            "doc_count_error_upper_bound" : 0,
            "sum_other_doc_count" : 0,
            "buckets" : [
              {
                "key" : "super_title",
                "doc_count" : 2
              }
            ]
          }
        }
      ]
    }
  }
}
```
</details>

<details> <summary> Without results </summary>

`title` aggregation does not exist

```
{
  "took" : 2,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 0,
      "relation" : "eq"
    },
    "max_score" : null,
    "hits" : [ ]
  },
  "aggregations" : {
    "genres" : {
      "doc_count_error_upper_bound" : 0,
      "sum_other_doc_count" : 0,
      "buckets" : [ ]
    }
  }
}
```


</details>

Reply:
A comment to that effect here (possibly even in Doc form) might help the reader understand that.

Reply:
Good idea. I'll add it.